### DISCUSSION TITLE
Persistent Data

### DISCUSSION THREAD

**Discussion by navascogt** [sree-hari-s/Expense-Tracker#21] [Ideas]
To have persistent data a database engine must be defined. I have some points to have in mind in order to do that:

1. Choice Structured or Non Structured Database
2. Be agnostic to the database, so we can implement with any flavor (PostgreSQL, MySQL, MariaDB, etc.)
3. Reviewing Streamlit  supports SQLAlchemy, it is a powerful ORM!
4. Where to place, in a container, Cloud provider, use Cloud based (RDS)?
5. For developer purposes it is recommended to use dockers for the app (Streamlit has articles to build it) and for the database.
6. Having the requirements and the scope to reach, with that design the database.

Maybe there are more points, but the above at least is for start the discussion to define the requirements.

**Comment by sree-hari-s** [comment]
Do you think a minimum viable app is ready currently?
